spacy/cli/train.py

```diff
@@ -295,7 +295,11 @@ def train_while_improving(
                 nlp.rehearse(raw_batch, sgd=optimizer, losses=losses, exclude=exclude)
         # TODO: refactor this so we don't have to run it separately in here
         for name, proc in nlp.pipeline:
-            if name not in exclude and hasattr(proc, "model"):
+            if (
+                name not in exclude
+                and hasattr(proc, "model")
+                and isinstance(proc.model, Model)
```

Comment:
We shouldn't do this `isinstance` check, if someone has a value set to the `model` attribute of their component that doesn't duck-type correctly, that's their bug.

We could check something like `not in (False, True, None)`. Or we could ask pipes to implement a `is_trainable` property.